build.gradle

```diff
@@ -15,6 +15,11 @@ plugins {
 group = 'software.amazon.cryptools'
 version = '2.3.2'
 ext.isFips = Boolean.getBoolean('FIPS')
+if (ext.isFips) {
+    ext.awsLcGitVersionId = 'AWS-LC-FIPS-2.0.0'
+} else {
+    ext.awsLcGitVersionId = 'v1.17.0'
```

Comment:
have we updated the submodule commit ID to match this?

Reply:
Updated it to the latest version, v1.17.1.

Reply:
It turns out that in v1.17.1, the RSA struct is opaque and ACCP has problem building with it. Upgrading to v1.17.1 will be done in another PR.

Reply:
Why did this else statement need to exist? This PR purports to only make changes to the FIPS build. Shouldn't the non-FIPS build continue to do everything as it used to do? I don't like that we now have to track the git version tag of LC to use in the gradle script, that seems really brittle and we're already seeing evidence of that in this thread.

Reply:
Ideally we want to stop using Git submodule to track our dependency on AWS-LC and use the values here as the source of truth. What's blocking us is a legacy build that has a hard time reaching out and pulling the right version of AWS-LC from Github. Once that's done, then we will drop submodule and only rely on this value.

Since we are using different version of AWS-LC for FIPS and non-FIPS builds, Git submodule is not a suitable solution.

Reply:
If that's the case, have we considered how this impacts our agility? AWS-LC iterates on these versions pretty infrequently based on a cursory glance at their [releases](https://github.com/aws/aws-lc/tags). Far less frequently then we pulled in submodule changes.

Reply:
The mechanism for updating submodule is the same as updating these values in the code: create a PR and get it merged.

Reply:
What I'm saying is that previously, you can point to /any/ AWS-LC commit. Now we're limited to whatever they decide to tag with a release. Though I suppose you could still use a commit hash as a git ref and achieve the same effect.

Reply:
As a matter of hygiene, I think it would be desirable to track LC by releases rather than individual commits (although it's good to have "break glass" options to reference individual commits). Using specific versions should help with stability, bug reports, etc.